### DISCUSSION TITLE
Custom Layouts

### DISCUSSION THREAD

**Discussion by AndreyKindin** [directus/directus#6542] [Archive]
### What version of Directus are you using?

v9.0.0-rc.81

### What version of Node.js are you using?

v14.16.1

### What database are you using?

mysql

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying Directus?

locally

### Describe the Bug

i took the example from 
https://docs.directus.io/guides/layouts/ 
and tried to make it work.

When PUBLIC_URL is set to "/"
i get the error messages in the log  

Route /extensions/layouts/admin/vue.089cc4dd.js doesn't exist.
request completed GET 404 /extensions/layouts/admin/vue.089cc4dd.js 7ms
Route /extensions/layouts/admin/@directus_extension-sdk.922e84b6.js does not exist.
request completed GET 404 /extensions/layouts/admin/@directus_extension-sdk.922e84b6.js 11ms

if i set it to https://.... url of directus there are no errors in directus log but in chrome console when trying to switch to custom layout 

Project Information
index.4c25381e.js:1 Version: v9.0.0-rc.81
index.4c25381e.js:1 Environment: production
vendor.1daab25f.js:1 Error: [useLayoutState]: This function has to be used inside a layout component.
vendor.1daab25f.js:1 TypeError: Cannot read property 'collection' of undefined

thanks for the help 

### To Reproduce

^^

**Comment by nickrum** [comment]
The issue when `PUBLIC_URL` is set to `/` has already been fixed.

Regarding the second issue, browsers use different contexts for modules imported from a domain with `https://` vs `http://`. I suspect this has something to do with that. What kind of TLS configuration are you using?
